### ISSUE TITLE
Add an additional configuration scope between 'default' and 'system' to accommodate HPC clusters

### ISSUE THREAD

**Issue by runiq** [LLNL/spack#43073]
### Summary

I would like to be able to put Spack's site config location (currently `$spack/etc`) somewhere else, outside of the spack build directory.

### Rationale

IMHO, configuration and code shouldn't go together. Currently, the site config resides right inside what I consider to be the spack project code.

Both system and user config scope locations are already configurable; the site config appears to be the only one that is not. Maybe I'm putting too much meaning into the "site" moniker, but "instance scope" might be a better fit than "site scope".

Moving the site config outside of the Spack tree would also help with updates. Currently, we update like this:

1.  Untar the release tarball to `/software/spack/<version>`
2. Copy contents of old `$spack/etc` into new `$spack/etc`, _but without touching `$spack/etc/defaults`_
3. Apply necessary changes for new spack version
4. Move symlink (`/software/spack/spack`) to point to the new version (`/software/spack/0.21.2`)

Especially the presence of `defaults` in the same directory is risky.

### Description

Similar in spirit to #26735, I'd like a variable `$SPACK_SITE_CONFIG_PATH`. If that is unset, Spack uses the current default (`$spack/etc`).

### General information

- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
Can't you use `SPACK_SYSTEM_CONFIG_PATH`? It defaults to `/etc/spack` -- even though adding `/etc/spack` system wide config is considered a mistake, it can be repurposed.

**Comment by runiq** [comment]
Yeah, sure, I could—but I figured that the system config was more for single hosts (i.e., a special node in a cluster) and not entire sites (i.e., the entire cluster). I also like the consistency of having variables for all three high-level config locations. :)

**Comment by runiq** [comment]
After reading through the docs again, it occurred to me that Spack's meaning of 'site' is different from mine. What Spack calls a 'site' is what I would call an 'instance'. In fact, [the docs call it that too](https://spack.readthedocs.io/en/latest/configuration.html#configuration-scopes):

> **site:** Stored in `$(prefix)/etc/spack/`. Settings here affect only _this instance_ of Spack

In my eyes, a _site_ is on a tier _higher than that of a single system_--say, an entire HPC cluster--and not lower.

Adding a config scope that sits between 'default' and 'system' would be a meaningful addition and should tie in with #43412 as well. I have changed the description accordingly. I'm not sure how to name that scope, however, given that 'site' is already taken. I propose renaming the existing 'site' scope to 'instance' and using 'site' for the scope between 'default' and 'system' instead, but I understand that changing established defaults is generally frowned upon. :thinking:
